lib/foreman_maintain/cli/service_command.rb

```diff
@@ -0,0 +1,125 @@
+module ForemanMaintain
+  module Cli
+    class ServiceCommand < Base
+      option '--exclude', "EXCLUDE", 'A comma-separated list of services to skip'
+      option '--only', "ONLY", 'A comma-separated list of services to include'
+
+      subcommand 'start', 'Start applicable services' do
+        # TODO: DRY up labels
+        label_option
+        tags_option
+        interactive_option
```

Comment:
what is the best way to not repeat these option methods each subcommand?

Reply:
Also do I need all of these? If I didn't include them I got an error (something about "assumeyes" not found iirc), but not sure if I  need all of them

Reply:
Currently, we don't have any way to avoid this repetition. If we want to achieve then we will need to add wrapper on top of it but I am not sure how easy it will be.

It is giving error because when we called method `run_scenario` from [base class](https://github.com/theforeman/foreman_maintain/blob/master/lib/foreman_maintain/cli/base.rb#L44) we have used assumeyes? method. 
Note - For flag options, Clamp provides "#{flag-option-name}?" method.

Reply:
@johnpmitsch do you need label and tags? It worked for for me when removed. The interactive_option(s) are needed by runner but I'm not sure if --assumeyes and --whitelist makes sense in the scripts. I'd suggest to fix the runner to not depend on these and use it only when available.

Back to your question - if not removed it could be included in common predecessor for the service subcommands. For such case there is another notation for subcommand declaration https://github.com/mdub/clamp#declaring-subcommands.

One thing I noticed and am not sure about is the declaration of --exclude and --only options. The way it is done it allows you to call `foreman-maintain service --only httpd status` and also `foreman-maintain service status --only httpd`. Both ways work both make sense but I'd stick with one to avoid confusion. While I like the first notation (reminds me sysv init scripts notation) it is the second one our CLI is using quite uniformly.

Reply:
@mbacovsky thanks, I can try to remove the label and tags option. I agree that `foreman-maintain service status --only httpd` would be good syntax and match our CLI usage already, I'll update to that :100: